gene2phenotype_project/gene2phenotype_app/views/publication.py

```diff
@@ -0,0 +1,99 @@
+from rest_framework import permissions, status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+
+from gene2phenotype_app.serializers import PublicationSerializer
+
+from gene2phenotype_app.models import Publication
+
+from .base import BaseAdd
+
+from ..utils import get_publication, get_authors
+
+
+"""
+    Retrieve publication data for a list of PMIDs.
+    If PMID is found in G2P then return details from G2P.
+    If PMID not found in G2P then returns info from EuropePMC.
+
+    Args:
+            (HttpRequest) request: HTTP request
+            (str) pmids: A comma-separated string of PMIDs
+
+    Returns:
+            Response object includes:
+                (list) results: contains publication data for each publication
+                                    - pmid
+                                    - title
+                                    - authors
+                                    - year
+                                    - source (possible values: 'G2P', 'EuropePMC')
+                (int) count: number of PMIDs
+    
+    Raises:
+            Invalid PMID
+"""
+@api_view(['GET'])
+def PublicationDetail(request, pmids):
+    id_list = pmids.split(',')
+    data = []
+    invalid_pmids = []
+
+    for pmid_str in id_list:
```

Comment:
Suggesting we change this , i do not think we need to use a try block in this case.
```
for pmid_str in id_list:
        if pmid_str.isdigit():
            pmid = int(pmid_str)
            try:
                publication = Publication.objects.get(pmid=pmid)
                data.append({
                    'pmid': int(publication.pmid),
                    'title': publication.title,
                    'authors': publication.authors,
                    'year': int(publication.year),
                    'source': 'G2P'
                })
            except Publication.DoesNotExist:
                # Query EuropePMC
                response = get_publication(pmid)
                if response['hitCount'] == 0:
                    invalid_pmids.append(pmid_str)
                else:
                    authors = get_authors(response)
                    year = None
                    publication_info = response['result']
                    title = publication_info['title']
                    if 'pubYear' in publication_info:
                        year = publication_info['pubYear']
                    data.append({
                        'pmid': int(pmid),
                        'title': title,
                        'authors': authors,
                        'year': int(year),
                        'source': 'EuropePMC'
                    })
        else:
            invalid_pmids.append(pmid_str)
```

Reply:
They are just different ways to do the same thing.